sql/analyzer/exec_builder.go

```diff
@@ -180,10 +180,18 @@ func (b *ExecBuilder) buildHashJoin(j *hashJoin, input sql.Schema, children ...s
 	if err != nil {
 		return nil, err
 	}
+	oldInJoin := false
+	if j.g.m.scope != nil {
```

Comment:
Rather than temporarily monkeying with the scope like this, why not pass a modified copy directly to this function?